### PR TITLE
Expose digestmod to allow custom selection of hash function

### PR DESCRIPTION
**Pull request by eyalfink**
preserving original behavior which uses hmac defaults.

### CODE REVIEW

**Comment by eyalfink** [comment]
I've signed it

File: oauth2client/xsrfutil.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def _force_bytes(s):
 
 
 @util.positional(2)
-def generate_token(key, user_id, action_id="", when=None):
+def generate_token(key, user_id, action_id="", when=None, digestmod=None):
```
Comment (nathanielmanistaatgoogle) [inline oauth2client/xsrfutil.py:50]: Don't you need to update the "Args:" section of the function doc string?

File: oauth2client/xsrfutil.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def generate_token(key, user_id, action_id="", when=None):
 
 
 @util.positional(3)
-def validate_token(key, token, user_id, action_id="", current_time=None):
+def validate_token(key, token, user_id, action_id="", current_time=None, digestmod=None):
```
Comment (nathanielmanistaatgoogle) [inline oauth2client/xsrfutil.py:82]: Doc string update here too?

Comment (eyalfink) [inline oauth2client/xsrfutil.py:82]: Good point.
Done.
PTAL.

**Comment by nathanielmanistaatgoogle** [comment]
Please squash commits - this change is simple enough that there's no need for it to be in two commits.

**Comment by nathanielmanistaatgoogle** [comment]
Also please add test coverage.

**Comment by jonparrott** [comment]
@eyalfink friendly ping. do you still want this to be merged?

You may ignore the request to squash commits, as github has that built-in now, but we do need test coverage.

**Comment by jonparrott** [comment]
@eyalfink another friendly ping. I'll close this PR in a few days if I don't hear from you.

**Pull request closed without merging**